Closes #111 dataset contents ref pages

vbshuliar opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

manciniedoardo commented:
> * A comment should be added to the `R/*.R` that these are generated and shouldn't be edited manually.
> * `README.md` should be updated to describe the new documentation process.

@vbshuliar once you update `README.md` could you also do the same in pharmaverseadam as we are now following the same process across both packages? thanks

vbshuliar commented:
@manciniedoardo @bundfussr Please take a look at the updated version.

manciniedoardo commented on README.md:
```diff
@@ -47,24 +47,51 @@ Then there are two main ways to extend the test data: either by adding new datas
+3. **Generate Documentation**:
+   - Use `roxygen2::roxygenize()` to update `.Rd` files in the `man/` directory.
+
+### Key Benefits of this Documentation Approach
```
Remove this section - unneeded for README.

manciniedoardo commented on README.md:
```diff
@@ -47,24 +47,51 @@ Then there are two main ways to extend the test data: either by adding new datas
-* The names and sources of test datasets are specified in `R/data.R`, for the purpose of generating documentation in the `man/` folder.
+* The names and sources of test datasets are specified in `R/*.R`, for the purpose of generating documentation in the `man/` folder.
+
+**Note:** The documentation process in `{pharmaversesdtm}` is automated for consistency and ease of maintenance. Metadata for each dataset, such as names, labels, descriptions, authors, and sources, is managed in a centralized CSV file (`inst/extdata/sdtms-specs.csv`) and used to generate `.R` documentation files. See the [Documentation Process](#documentation-process) for details.
```
Why is this a `.csv` while the one in pharmaverseadam is a `.xlsx`? Can we align?

manciniedoardo replied:
Thanks for adding this Note. Would you please also take the opportunity to review the bullet-pointed list in this section and ensure it's still up-to-date? For instance, we should remove references to renv.

manciniedoardo commented on README.md:
```diff
@@ -47,24 +47,51 @@ Then there are two main ways to extend the test data: either by adding new datas
 
+# Documentation Process
+
+The documentation process in `{pharmaversesdtm}` is automated for consistency and ease of maintenance. Metadata for each dataset, such as names, labels, descriptions, authors, and sources, is managed in a centralized CSV file (`inst/extdata/sdtms-specs.csv`) and used to generate `.R` documentation files.
```
Again, why CSV?

manciniedoardo reviewed (changes requested):
Thanks @vbshuliar - I tested the actual process and looks great. I've added a couple of comments.

In addition to the above I also tried rendering the site. Looks good, but can you please restructure the reference page so that it mentions datasets and not functions:

![image](https://github.com/user-attachments/assets/c69cdf9a-3f5a-4c21-bcf7-3c28e1fd6c77)

vbshuliar commented:
I updated the layout as you requested. Please, proceed with the rereview @manciniedoardo 

![image](https://github.com/user-attachments/assets/34c47353-ff0b-4b42-9609-c62a3256e068)
![image](https://github.com/user-attachments/assets/71b1a353-1c84-409d-b8ec-11f69764b374)

bundfussr commented on inst/extdata/sdtms-specs.xlsx:
Could we use a JSON file?

Then it is possible to see changes in github.

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,134 @@
+# Load the XLSX file with metadata
```
Could you add a header describing the purpose of the script?

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,134 @@
+
+    # Create documentation for the current dataset
+    doc_string <- paste(
+      sprintf("# This file is automatically generated."),
```
Could we add more information? For example:
```suggestion
      sprintf("# This file is automatically generated by data-raw/create_sdtms_data.R."),
      sprintf("# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R."),
```

bundfussr reviewed (changes requested):
Could you update such that the labels are displayed on the reference page? For example:
![image](https://github.com/user-attachments/assets/77a2ad19-3c17-45d6-aaea-7fffbd5f51df)

vbshuliar commented:
- Added the table of contents for the readme.
- Replaced XLSX with JSON.
- Fixed the reference page to display titles correctly (instead of file names).

    <img src="https://github.com/user-attachments/assets/cd582bdc-3a69-4558-8ea5-d707cdbff6c6" alt="Image 1" width="500">
    <img src="https://github.com/user-attachments/assets/41a414a3-ce81-4b76-a152-9f727bc5b7e7" alt="Image 2" width="500">
    <img src="https://github.com/user-attachments/assets/17bbfeb0-79d4-47a2-9a46-f4938e6c9442" alt="Image 3" width="500">

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,142 @@
+}
+
+# Helper function to check the length of a string and append " # nolint" if it exceeds 90 characters
+check_and_append_nolint <- function(text) {
+  if (nchar(text) > 80) {
+    text <- paste0(text, " # nolint")
+  }
+  return(text)
+}
```
This doesn't work properly because the "# nolint" appears in the rendered documentation:
![image](https://github.com/user-attachments/assets/c0aeb8ed-c806-4331-bb59-0ae1fa1bd57f)

bundfussr reviewed (changes requested):
@vbshuliar , thanks for the updates.

Could you make the URLs in the HTML version clickable? For example, the yellow highlighted URL below is just text.
![image](https://github.com/user-attachments/assets/0f5d8df6-2436-4b6e-9023-958b7c8d3f61)

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,156 @@
+check_and_break_line <- function(text, max_length = 80) {
+  if (nchar(text) > max_length) {
+    # Split the text into chunks of at most max_length characters
+    split_text <- strsplit(text, "(?<=.{1," , max_length, "})(\\s|\\n)", perl = TRUE)[[1]]
```
This doesn't work (see lintr check). What about using `stringr::str_wrap()`?

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,156 @@
+    # Check and adjust lengths of metadata fields
+    dataset_label <- check_and_break_line(dataset_label)
+    dataset_description <- check_and_break_line(dataset_description)
+    dataset_source <- check_and_break_line(generate_hyperlink(dataset_source, dataset_label))
```
This makes no sense if `dataset_source` doesn't contains an URL.

I would remove `generate_hyperlink()` and use mark-up in the JSON file. For example:
```
    {
      "name": "ae",
      "label": "Adverse Events",
      "description": "An updated SDTM AE dataset that uses the CDISC pilot project",
      "source": "[https://github.com/pharmaverse/pharmaversesdtm/blob/main/data/ae.rda]",
      "author": "Gopi Vegesna"
    },
```
or
```
    {
      "name": "ae",
      "label": "Adverse Events",
      "description": "An updated SDTM AE dataset that uses the CDISC pilot project",
      "source": "[pharmaversesdtm::ae](https://github.com/pharmaverse/pharmaversesdtm/blob/main/data/ae.rda)",
      "author": "Gopi Vegesna"
    },
```

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,156 @@
+      sprintf("# This file is automatically generated by data-raw/create_sdtms_data.R."),
+      sprintf("# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R."),
+      sprintf("# Manual edits are not recommended, as changes may be overwritten."),
+      sprintf("#' %s", dataset_label),
```
This will fail once `check_and_break_line()` works because only the first line starts with `#'` then.

bundfussr commented on .lintr:
```diff
@@ -7,8 +7,8 @@ linters: linters_with_defaults(
 exclusions: list(
-    "R/data.R" = Inf,
+    "R" = Inf,
     "inst" = list(undesirable_function_linter = Inf),
     "vignettes" = list(undesirable_function_linter = Inf),
-    "data-raw" = list(undesirable_function_linter = Inf)
+    "data-raw" = Inf
```
I'm not sure if we should turn off lintr completely for all files in `R` and `data-raw`.

@manciniedoardo , what do you think?

manciniedoardo replied:
ideally not completely, is there a more granular approach?

manciniedoardo replied:
that's a good idea

manciniedoardo reviewed (approved):
LGTM - @bundfussr please add your review when you can so that I can then do a release.

bundfussr commented on _pkgdown.yml:
```diff
@@ -13,13 +13,66 @@ repo:
+  - title: "Datasets"
+    desc: "List of all available datasets."
+    contents:
+      - ae
```
We should add `@keywords dataset` to the `*.R` files (except `get_terms.R`) and use 
```
      - has_keyword("dataset")
```
here.
Then we don't need to maintain this list.

bundfussr commented on inst/extdata/sdtms-specs.json:
```diff
@@ -0,0 +1,292 @@
+    {
+      "name": "ae_ophtha",
+      "label": "Adverse Events for Ophthalmology",
+      "description": "An example Adverse Events SDTM dataset with ophthalmology-specific variable AELAT",
```
```suggestion
      "description": "An example Adverse Events SDTM dataset with ophthalmology-specific variable `AELAT`",
```

bundfussr commented on inst/extdata/sdtms-specs.json:
```diff
@@ -0,0 +1,292 @@
+    {
+      "name": "ex_ophtha",
+      "label": "Exposure for Ophthalmology",
+      "description": "An example Exposure SDTM dataset with ophthalmology-specific variables such as EXLOC and EXLAT",
```
```suggestion
      "description": "An example Exposure SDTM dataset with ophthalmology-specific variables such as `EXLOC` and `EXLAT`",
```

bundfussr commented on inst/extdata/sdtms-specs.json:
```diff
@@ -0,0 +1,292 @@
+    {
+      "name": "supprs_onco_imwg",
+      "label": "Supplemental Qualifiers for RS_ONCO_IMWG",
+      "description": "A SDTM supplemental RS dataset using IMWG criteria intended to be used with rs_onco_imwg",
```
```suggestion
      "description": "A SDTM supplemental RS dataset using IMWG criteria intended to be used with `rs_onco_imwg`",
```

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,129 @@
+    sprintf("#' @name %s", dataset_name),
+    sprintf("#' @title %s", dataset_label),
+    sprintf("#' @keywords dataset"),
+    sprintf("#' @description dataset"),
```
This looks strange:
![image](https://github.com/user-attachments/assets/7191e0d3-bdc7-451b-a69f-2ef1b582f465)

bundfussr commented on data-raw/create_sdtms_data.R:
```diff
@@ -0,0 +1,129 @@
+  metadata <- specs[specs$name == dataset_name, ]
+
+  if (nrow(metadata) == 0) {
+    cat("No metadata found for", dataset_name, "- using default values.\n")
```
Should this be a warning or an error?

The documentation of `rs_onco_ca125` and `supprs_onco_ca125` was removed but it was not noticed.

I've re-added it.

bundfussr reviewed (changes requested):
The metadata for `vs_peds` was lost. It is available at https://pharmaverse.github.io/pharmaversesdtm/reference/vs_peds.html but in this branch I get
```
Warning message:
No metadata found for vs_peds - using default values. 
```
when calling `create_sdtms_data.R`.

bundfussr reviewed (approved):
Looks good now.

@vbshuliar , thanks for your great work and patience!
